include/modem/modem_info.h

```diff
@@ -61,10 +70,19 @@ extern "C" {
  * a conversion to dBm and dB respectively. See modem AT
  * command reference guide for more information.
  */
-#define RSRP_IDX_TO_DBM(rsrp) ((rsrp) - RSRP_OFFSET_VAL)
+#define RSRP_IDX_TO_DBM(rsrp) ((rsrp) < 0 ?                        \
+			       (rsrp) - RSRP_OFFSET_VAL_NEGATIVE : \
+			       (rsrp) - RSRP_OFFSET_VAL_POSITIVE)
 
-#define RSRQ_IDX_TO_DB(rsrq) ((((float)(rsrq)) * RSRQ_SCALE_VAL) - \
-			      RSRQ_OFFSET_VAL)
+/** Modem returns RSRP and RSRQ as index values which require
```

Comment:
Please modify comment to only describe RSRQ conversion and do the same for the RSRP macro comment.